src/common/eeprom.cpp

```diff
@@ -551,48 +561,38 @@ static uint16_t eeprom_var_size(enum eevar_id id) {
     return 0;
 }
 
-static uint16_t eeprom_var_addr(enum eevar_id id) {
-    uint16_t addr = EEPROM_ADDRESS;
+static uint16_t eeprom_var_addr(enum eevar_id id, uint16_t addr) {
```

Comment:
I agree that intptr_t would be better, but I wanted to make as few changes as possible, because if I started to change parts of eeprom that I do not like, I would rewrite it all